recipes/get_cluster_json.rb

```diff
@@ -0,0 +1,19 @@
+# frozen_string_literal: true
+
+#
+# Cookbook:: aws-parallelcluster-test
+# Recipe:: test_nvidia
+#
+# Copyright:: 2013-2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance with the
+# License. A copy of the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "LICENSE.txt" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES
+# OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions and
+# limitations under the License.
+
+cluster_json = JSON.generate(node['cluster'], allow_nan: true, max_nesting: false)
+puts "CLUSTER JSON: #{cluster_json}"
```

Comment:
I am writing the configuration state out with this recipe so that we can call it from the image validation tests to get it as a JSON structure. 
Currently, the tests get the configuration values by grepping the defaults.rb file. This works for simple key value pairs, but in some cases these values have a more complex calculation (e.g. value_for_platform), so a simple grep won't work.
By outputting the config a JSON string, the tests can use `jq` to query the config tree where all the values are fully resolved.